src/writer.rs

```diff
@@ -297,6 +320,34 @@ impl<'a> WriterOptions<'a> {
     }
 }
 
+fn write_extensions<W, E>(
+    mut writer: ByteOrdered<W, E>,
+    extensions: Option<&ExtensionSequence>,
+) -> Result<()>
+where
+    W: Write,
+    E: Endian,
+{
+    let extensions = match extensions {
+        Some(extensions) => extensions,
+        None => { writer.write_u32(0)?; return Ok(())},
```

Comment:
Could you please configure your IDE to format on save? There's no automatic formatting in our cicd or elsewhere. rustfmt is here to avoid that kind of discussion :)